articles/cognitive-services/Bing-Web-Search/quickstarts/nodejs.md

```diff
@@ -22,102 +22,145 @@ Use this quickstart to make your first call to the Bing Web Search API and recei
 Here are a few things that you'll need before running this quickstart:
 
 * [Node.js 6](https://nodejs.org/en/download/) or later
-* A subscription key  
+* A subscription key
 
 ## Create a project and declare required modules
 
-Create a new Node.js project in your favorite IDE or editor. Then copy the code snippet below into your project. This quickstart uses strict mode and requires the `https` module to send and receive data.
+Create a new Node.js project in your favorite IDE or editor.
+Then copy the code snippet below into your project in a file named `search.js`.
+First we require `https` from the standard library, we'll use it to send and receive data.
 
 ```javascript
-// Use strict mode.
-'use strict';
+// Usage:
+//
+//   node search.js "My awesome query"
+//
+const https = require('https')
+```
+
+## Set the subscription key
+
+This code looks up the environment variable, `AZURE_SUBSCRIPTION_KEY`,
+which you can find on
+[your apis page](https://azure.microsoft.com/en-us/try/cognitive-services/my-apis/?apiSlug=search-api-v7).
 
-// Require the https module.
-let https = require('https');
+```javascript
+// HOWTO: set an environment variable https://is.gd/zu55jA
+const SUBSCRIPTION_KEY = process.env['AZURE_SUBSCRIPTION_KEY']
+if (!SUBSCRIPTION_KEY) {
+  throw new Error('Missing the AZURE_SUBSCRIPTION_KEY environment varable')
+}
 ```
 
-## Define variables
 
-A few variables must be set before we can continue. Confirm that the `host` and `path` are valid and replace the `subscriptionKey` value with a valid subscription key from your Azure account. Feel free to customize the search query by replacing the value for `term`.
+
+## Create a function to make the request
+
+This function will make a secure GET request, saving the search query as a query parameter in the path.
+We use `encodeURIComponent` to escape any invalid characters, and we pass our subscription key in a header.
+The callback receives a
+[response](https://nodejs.org/dist/latest-v10.x/docs/api/http.html#http_class_http_serverresponse),
+subscribing to the `data` event in order to aggregate the JSON body,
+the `error` event to log any issues,
+and the `end` event so it knows when it has the full body.
```

Comment:
Edit: and the `end` event to know when the message should be considered complete.